nixos/modules/image/repart.nix

```diff
@@ -95,6 +95,40 @@ in
       '';
     };
 
+    compression = lib.mkOption {
+      description = "Image compression";
+      default = { };
+      type = lib.types.submodule ({ config, ... }: {
+        options = {
+          enable = lib.mkEnableOption (lib.mdDoc "Image compression");
+
+          algorithm = lib.mkOption {
+            type = lib.types.enum [ "zstd" "xz" ];
+            default = "zstd";
+            description = "Compression algorithm";
+          };
+
+          level = lib.mkOption {
+            type = lib.types.str;
+            description = ''
+              Compression level.
+
+              This is a string so that you can set the appropriate level specific to the algorithm.
+            '';
+          };
+        };
+        config = {
+          # Generally default to slightly faster than default compression
+          # levels under the assumption that most of the building will be done
+          # for development and release builds will be customized.
+          level = lib.mkDefault {
```

Comment:
I'm assuming we set the default compression level here instead of in the option definition since the value that ends up getting passed to the compression tool ends up depending on the tool itself. Since we don't set the default value in the option, can we use `mkOptionDefault` here to allow users to still use `mkDefault`?